odoo/addons/base/models/ir_model.py

```diff
@@ -2149,13 +2171,11 @@ def delete(records):
         constraints = self.env['ir.model.constraint'].search([('module', 'in', modules.ids)])
         constraints._module_data_uninstall()
 
-        # Remove fields, selections and relations. Note that the selections of
-        # removed fields do not require any "data fix", as their corresponding
-        # column no longer exists. We can therefore completely ignore them. That
-        # is why selections are removed after fields: most selections are
-        # deleted on cascade by their corresponding field.
-        delete(self.env['ir.model.fields'].browse(unique(field_ids)))
+        # ir.model.fields.selection must be deleted first and before any fields, SQL CASCADE
+        # deletes must be avoided if the ondelete actions are to be executed on the records
+        # pointing to the to-be-deleted ir.model.fields.selection
```

Comment:
This is very obscure.  You could explain with an example.
Let me give it a try, to see whether I understood the issue.
> If we delete a selection field, and some of its values have `ondelete='cascade'`, we expect the records with that value to be deleted.  If we delete the field first, the column is dropped and the selection is gone, and thus the records above will not be deleted.